src/test/kotlin/com/pubnub/api/legacy/endpoints/pubsub/PublishTest.kt

```diff
@@ -562,4 +534,84 @@ class PublishTest : BaseTest() {
         assertEquals("0", requests[0].queryParameter("store").firstValue())
         assertFalse(requests[0].queryParameter("ttl").isPresent)
     }
+
+    @Test
+    fun testSpaceIdQueryParamIsPassedInPublish() {
+        val spaceIdValue = "thisIsSpaceId"
+
+        stubFor(
+            get(urlPathEqualTo("/publish/myPublishKey/mySubscribeKey/0/coolChannel/0/%22hi%22"))
+                .willReturn(aResponse().withBody("""[1,"Sent","15883272000000000"]"""))
+        )
+
+        pubnub.publish(
+            channel = "coolChannel",
+            message = "hi",
+            spaceId = SpaceId(spaceIdValue)
+        ).sync()!!
+
+        val requests = findAll(getRequestedFor(urlMatching("/.*")))
+        assertEquals(1, requests.size)
+        assertThat(
+            requests[0].queryParams.map { (k, v) -> k to v.firstValue() }.toMap(),
+            hasEntry(Publish.SPACE_ID_QUERY_PARAM, spaceIdValue)
+        )
+    }
+
+    @Test
+    fun testMissingSpaceIdQueryParamIsNotSet() {
+        stubFor(
+            get(urlPathEqualTo("/publish/myPublishKey/mySubscribeKey/0/coolChannel/0/%22hi%22"))
+                .willReturn(aResponse().withBody("""[1,"Sent","15883272000000000"]"""))
+        )
+
+        pubnub.publish(
+            channel = "coolChannel",
+            message = "hi"
+        ).sync()!!
+
+        val requests = findAll(getRequestedFor(urlMatching("/.*")))
+        assertEquals(1, requests.size)
+        assertThat(requests[0].queryParams, not(hasKey(Publish.SPACE_ID_QUERY_PARAM)))
+    }
+
+    @Test
+    fun testMessageTypeQueryParamIsPassedInPublish() {
+        val messageTypeValue = "thisIsSpaceId"
+
+        stubFor(
+            get(urlPathEqualTo("/publish/myPublishKey/mySubscribeKey/0/coolChannel/0/%22hi%22"))
```

Comment:
What about extracting "coolChannel" to variable. Then the variable can be used here and in publish call:
channel = "coolChannel",
?

Reply:
Sure. Why not